mailet/standard/src/test/java/org/apache/james/transport/mailets/MimeDecodingMailetTest.java

```diff
@@ -0,0 +1,131 @@
+package org.apache.james.transport.mailets;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.Map;
+
+import javax.mail.MessagingException;
+
+import org.apache.mailet.Mail;
+import org.apache.mailet.MailetContext;
+import org.apache.mailet.MailetException;
+import org.apache.mailet.base.test.FakeMail;
+import org.apache.mailet.base.test.FakeMailContext;
+import org.apache.mailet.base.test.FakeMailetConfig;
+import org.assertj.core.data.MapEntry;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.slf4j.Logger;
+
+import com.google.common.base.Charsets;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+public class MimeDecodingMailetTest {
+
+    private static final String MAIL_ATTRIBUTE = "mime.attachments";
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    private Logger logger;
+    private MailetContext mailetContext;
+    private MimeDecodingMailet testee;
+
+    @Before
+    public void setUp() throws Exception {
+        testee = new MimeDecodingMailet();
+        logger = mock(Logger.class);
+        mailetContext = FakeMailContext.builder()
+                .logger(logger)
+                .build();
+    }
+
+    @Test
+    public void initShouldThrowWhenNoAttributeParameter() throws MessagingException {
+        FakeMailetConfig mailetConfig = FakeMailetConfig.builder()
+                .mailetName("Test")
+                .mailetContext(mailetContext)
+                .build();
+        expectedException.expect(MailetException.class);
+        testee.init(mailetConfig);
+    }
+
+    @Test
+    public void serviceShouldThrowWhenAttributeContentIsNotAMap() throws MessagingException {
+        FakeMailetConfig mailetConfig = FakeMailetConfig.builder()
+                .mailetName("Test")
+                .mailetContext(mailetContext)
+                .setProperty(MimeDecodingMailet.ATTRIBUTE_PARAMETER_NAME, MAIL_ATTRIBUTE)
+                .build();
+        testee.init(mailetConfig);
+
+        Mail mail = mock(Mail.class);
+        when(mail.getAttribute(MAIL_ATTRIBUTE)).thenReturn(ImmutableList.of());
+
+        expectedException.expect(MailetException.class);
+        testee.service(mail);
+    }
+
+    @Test
+    public void serverShouldSetAttributeWhenNone() throws MessagingException {
+        FakeMailetConfig mailetConfig = FakeMailetConfig.builder()
+                .mailetName("Test")
+                .mailetContext(mailetContext)
+                .setProperty(MimeDecodingMailet.ATTRIBUTE_PARAMETER_NAME, MAIL_ATTRIBUTE)
+                .build();
+        testee.init(mailetConfig);
+
+        Mail mail = mock(Mail.class);
+
+        testee.service(mail);
+        assertThat(mail.getAttribute(MAIL_ATTRIBUTE)).isNull();
```

Comment:
I think the idea is that it was set before and it unsetted by the mailet if I understood correctly